developers/weaviate/client-libraries/typescript.mdx

```diff
@@ -0,0 +1,162 @@
+---
+title: TypeScript
+sidebar_position: 2
+image: og/docs/client-libraries.jpg
+# tags: ['TypeScript', 'client library']
+---
+import Badges from '/_includes/badges.mdx';
+
+<Badges/>
+
+:::note TypeScript client version
+The current TypeScript client version is `v||site.typescript_client_version||`.
+:::
+
+## Installation and setup
+
+The TypeScript client library package can be installed using [npm](https://www.npmjs.com/).
+
+<!-- Replace $ .. examples to remove the prompt ($) as it gets copied too along with the actual command -->
+```bash
+npm install weaviate-ts-client
+```
+
+## Usage and type definitions
+
+### Usage
+
+Once the client has been installed, you can use the client in your TypeScript scripts, such as in the following example.
+
+```ts
+import weaviate, { WeaviateClient } from 'weaviate-ts-client';
+
+const client: WeaviateClient = weaviate.client({
+  scheme: "https",
+  host: "some-endpoint.weaviate.network",
+});
+
+client
+  .schema
+  .getter()
+  .do()
+  .then((res: any) => {
+    console.log(res);
+  })
+  .catch((err: Error) => {
+    console.error(err)
+  });
+```
+
+### Type definitions
+
+The type definitions can be found under the `types` subdirectory of the package in the `*.d.ts` files, for example as shown on the [npm package page](https://www.npmjs.com/package/weaviate-ts-client?activeTab=code).
+
+## Authentication
+
+import ClientAuthIntro from '/developers/weaviate/client-libraries/_components/client.auth.introduction.mdx'
+
+<ClientAuthIntro clientName="TypeScript"/>
+
+### WCS authentication
+
+import ClientAuthWCS from '/developers/weaviate/client-libraries/_components/client.auth.wcs.mdx'
+
+<ClientAuthWCS />
+
+### Resource Owner Password Flow
+
+import ClientAuthFlowResourceOwnerPassword from '/developers/weaviate/client-libraries/_components/client.auth.flow.resource.owner.password.mdx'
+
+<ClientAuthFlowResourceOwnerPassword />
+
+```ts
+const client: WeaviateClient = weaviate.client({
+  scheme: "https",
+  host: "some-endpoint.weaviate.network",
+  authClientSecret: new weaviate.AuthUserPasswordCredentials({
+    username: "user123",
+    password: "password",
+    scopes: ["offline_access"]  // optional, depends on the configuration of your identity provider (not required with WCS)
+  })
+});
```

Comment:
This will apply for all the following auth/oidc code samples here and below:

With the merging of [this PR](https://github.com/weaviate/typescript-client/pull/32), we can import all types alongside the client default import:

```typescript
import weaviate, { AuthUserPasswordCredentials } from 'weaviate-ts-client';
```

this means that we can use the constructor without the `weaviate` qualifier:

```typescript
const client: WeaviateClient = weaviate.client({
   scheme: "https",
   host: "some-endpoint.weaviate.network",
   authClientSecret: new AuthUserPasswordCredentials({
     username: "user123",
     password: "password",
     scopes: ["offline_access"]  // optional, depends on the configuration of your identity provider (not required with WCS)
   })
 });
```

Reply:
Perfect. Updated here & in WCS docs.